backend/src/core/service/user.py

```diff
@@ -0,0 +1,114 @@
+from typing import Annotated
+from fastapi import Depends, Request, Response, HTTPException
+
+from ..models.user import UserCreateRequest, UserCreateResponse, UserLoginRequest, UserLoginResponse, UserSessionResponse, UserInDB
+from ..utils.datetime import DateTimeUtils
+from ..sql.operations.user import UserOperation
+from .validators.user_validator import UserValidator
+from ..utils.token import TokenUtils
+from .helpers.user_helper import UserHelper
+
+
+class UserService():
+    def __init__(
+        self,
+        user_opr: UserOperation = Depends()
+    ):
+        self.user_opr = user_opr
+
+    async def create_user(self, user_create: UserCreateRequest) -> UserCreateResponse:
+        """Create a new user with hashed password."""
+        # Validate user input
+        UserValidator.validate_email(user_create.email)
```

Comment:
Why do we throw in UserValidator? would it be more reusable if validate return Boolean then we throw in the service instead?